mesh/mesh.cpp

```diff
@@ -12050,28 +12052,32 @@ FaceGeometricFactors::FaceGeometricFactors(const Mesh *mesh,
                                           ElementDofOrdering::LEXICOGRAPHIC,
                                           type,
                                           L2FaceValues::SingleValued );
-   Vector Fnodes(face_restr->Height());
+   Vector Fnodes(face_restr->Height(), d_mt);
    face_restr->Mult(*nodes, Fnodes);
 
    unsigned eval_flags = 0;
+
+   MemoryType my_d_mt = (d_mt != MemoryType::DEFAULT) ? d_mt :
```

Comment:
I have a preference for less generic names than `my_***`, but I see that this file is already using such prefix, so I'm fine with it.